alibi_detect/utils/saving.py

```diff
@@ -1089,21 +1073,20 @@ def load_tf_model(filepath: str,
     -------
     Loaded model.
     """
-    if not load_dir:
-        model_dir = os.path.join(filepath, 'model')
-    else:
-        model_dir = os.path.join(filepath, load_dir)
-    if not os.path.isdir(model_dir):
+    model_dir = Path(filepath).joinpath(load_dir)
+    # Check if path exists
+    if not model_dir.is_dir():
         logger.warning('Directory {} does not exist.'.format(model_dir))
         return None
-    if model_name + '.h5' not in [f for f in os.listdir(model_dir) if not f.startswith('.')]:
+    # Check if model exists
+    if model_name + '.h5' not in [f.name for f in model_dir.glob('[!.]*.h5')]:
```

Comment:
Not sure this is correct, what are we matching?

Reply:
`[f for f in os.listdir(model_dir) if not f.startswith('.')]` returned all files in `model_dir` that don't begin with `.`, and then we checked if `model_name + '.h5` existed within that list. 

`[f.name for f in model_dir.glob('[!.]*.h5')]` should also be returning all files that don't begin with `.`, but with the added restriction of only returning files ending in `.h5`. Idea behind the added restriction was that the `if model_name + '.h5' not in [...]` would be searching through a shorter list. I did test both in a quick colab notebook but guess further tests might be helpful here.

Reply:
Just putting a quick notebook together to test this further...

Reply:
Oh I was thinking this was regexp but different rules for globbing, so it does what we want!

Reply:
Ta da!

https://colab.research.google.com/drive/17Iw819bN-mwuorOpYw15x1bIdYa4OIuj?usp=sharing